File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int[] arr = new int[3];

            arr[0] = sc.nextInt();
            arr[1] = sc.nextInt();
            arr[2] = sc.nextInt();
            Arrays.sort(arr);

            if (arr[0] == arr[2] && arr[0] == 0) {
                break;
            }

            if (arr[0] + arr[1] <= arr[2]) {
                System.out.println("Invalid");
            } else {
                if (arr[0] == arr[2]) {
                    System.out.println("Equilateral");
                } else if (arr[0] == arr[1] && arr[1] != arr[2] || arr[1] == arr[2] && arr[0] != arr[1]) {
                    System.out.println("Isosceles");
                } else if (arr[0] != arr[1] && arr[1] != arr[2]) {
                    System.out.println("Scalene");
                }
            }
        }
    }
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

# sys.stdin = open("input.txt", "r")

while(True):
    arr = list(map(int, sys.stdin.readline().split()))
    arr = sorted(arr)

    if arr[0] == arr[2] == 0:
        break

    if arr[0] + arr[1] <= arr[2]:
        print('Invalid')
    else:
        if arr[0] == arr[2]:
            print('Equilateral')
        elif (arr[0] == arr[1] and arr[1] != arr[2]) or (arr[0] != arr[1] and arr[1] == arr[2]):
            print('Isosceles')
        else:
            print('Scalene')